docs/source/notebooks/Density_matrix_Fock_state.ipynb

```diff
@@ -0,0 +1,487 @@
+       "<path d=\"M125,75 L225,75\" stroke=\"darkred\" stroke-width=\"3\" fill=\"none\" />\n",
+       "<path d=\"M125,125 L225,125\" stroke=\"darkred\" stroke-width=\"3\" fill=\"none\" />\n",
+       "<path d=\"M125,175 L225,175\" stroke=\"darkred\" stroke-width=\"3\" fill=\"none\" />\n",
+       "<path d=\"M125,225 L225,225\" stroke=\"darkred\" stroke-width=\"3\" fill=\"none\" />\n",
+       "<path d=\"M125,275 L225,275\" stroke=\"darkred\" stroke-width=\"3\" fill=\"none\" />\n",
+       "<path d=\"M127.5,2.5 L222.5,2.5 L222.5,297.5 L127.5,297.5 Z\" stroke=\"darkred\" fill=\"lightpink\" stroke-linejoin=\"miter\" />\n",
+       "<text x=\"135\" y=\"24\" font-size=\"8\" text-anchor=\"start\" font-weight=\"bold\"><tspan x=\"135\" dy=\"0em\">POSTPROCESSED</tspan><tspan x=\"135\" dy=\"1em\">CNOT</tspan></text>\n",
+       "<path d=\"M225,25 L240,25\" stroke-width=\"3\" stroke=\"darkred\" stroke-linejoin=\"miter\" fill=\"none\" />\n",
+       "<path d=\"M225,75 L240,75\" stroke-width=\"3\" stroke=\"darkred\" stroke-linejoin=\"miter\" fill=\"none\" />\n",
+       "<path d=\"M225,125 L240,125\" stroke-width=\"3\" stroke=\"darkred\" stroke-linejoin=\"miter\" fill=\"none\" />\n",
+       "<path d=\"M225,175 L240,175\" stroke-width=\"3\" stroke=\"darkred\" stroke-linejoin=\"miter\" fill=\"none\" />\n",
+       "<path d=\"M225,225 L240,225\" stroke-width=\"3\" stroke=\"darkred\" stroke-linejoin=\"miter\" fill=\"none\" />\n",
+       "<path d=\"M225,275 L240,275\" stroke-width=\"3\" stroke=\"darkred\" stroke-linejoin=\"miter\" fill=\"none\" />\n",
+       "<text x=\"250\" y=\"28\" font-size=\"7\" text-anchor=\"end\">0</text>\n",
+       "<text x=\"250\" y=\"78\" font-size=\"7\" text-anchor=\"end\">1</text>\n",
+       "<text x=\"250\" y=\"128\" font-size=\"7\" text-anchor=\"end\">2</text>\n",
+       "<text x=\"250\" y=\"178\" font-size=\"7\" text-anchor=\"end\">3</text>\n",
+       "<text x=\"250\" y=\"228\" font-size=\"7\" text-anchor=\"end\">4</text>\n",
+       "<text x=\"250\" y=\"278\" font-size=\"7\" text-anchor=\"end\">5</text>\n",
+       "<text x=\"0\" y=\"28\" font-size=\"7\" text-anchor=\"start\">0</text>\n",
+       "<text x=\"0\" y=\"78\" font-size=\"7\" text-anchor=\"start\">1</text>\n",
+       "<text x=\"0\" y=\"128\" font-size=\"7\" text-anchor=\"start\">2</text>\n",
+       "<text x=\"0\" y=\"178\" font-size=\"7\" text-anchor=\"start\">3</text>\n",
+       "<text x=\"0\" y=\"228\" font-size=\"7\" text-anchor=\"start\">4</text>\n",
+       "<text x=\"0\" y=\"278\" font-size=\"7\" text-anchor=\"start\">5</text>\n",
+       "</svg>"
+      ],
+      "text/plain": [
+       "<drawsvg.drawing.Drawing at 0x7f643a478210>"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "bell_circ = Circuit(m=6)\n",
+    "\n",
+    "bell_circ.add(0, BS.H())\n",
+    "bell_circ.add(0, catalog['postprocessed cnot'].build_circuit())\n",
```

Comment:
I'd put `merge=True` as a parameter of the `add` function for better display (no change in the computation)